src/core/main.c

```diff
@@ -164,6 +164,42 @@ static char **saved_env = NULL;
 static int parse_configuration(const struct rlimit *saved_rlimit_nofile,
                                const struct rlimit *saved_rlimit_memlock);
 
+static int manager_find_user_config_paths(char **ret_file, char ***ret_dirs) {
+        _cleanup_free_ char *file = NULL, *config = NULL;
+        _cleanup_strv_free_ char **dirs = NULL;
+        int r;
+
+        r = xdg_user_config_dir(&config, "/systemd");
+        /* ENXIO means XDG_RUNTIME_DIR is not set, don't consider that fatal. */
+        if (r < 0 && r != -ENXIO)
+                return r;
+
+        file = path_join(config, "user.conf");
+        if (!file)
+                return -ENOMEM;
+
+        if (access(file, F_OK) < 0 &&
```

Comment:
hmm, this is ugly... access() checking first and opening then is ugly...

please just open the file right away, and if you get ENOENT try the next one